Library/ViewModels/CategorySelectionHeaderViewModel.swift

```diff
@@ -0,0 +1,76 @@
+import Foundation
+import ReactiveSwift
+
+public enum HeaderViewContext {
+  case categorySelection
+  case curatedProjects
+}
+
+public protocol CategorySelectionHeaderViewModelInputs {
+  func configure(with context: HeaderViewContext)
+}
+
+public protocol CategorySelectionHeaderViewModelOutputs {
+  var stepLabelText: Signal<String, Never> { get }
+  var subtitleLabelText: Signal<String, Never> { get }
+  var titleLabelText: Signal<String, Never> { get }
+}
+
+public protocol CategorySelectionHeaderViewModelType {
+  var inputs: CategorySelectionHeaderViewModelInputs { get }
+  var outputs: CategorySelectionHeaderViewModelOutputs { get }
+}
+
+public final class CategorySelectionHeaderViewModel: CategorySelectionHeaderViewModelType,
+  CategorySelectionHeaderViewModelInputs, CategorySelectionHeaderViewModelOutputs {
+  public init() {
+    self.stepLabelText = self.contextSignal
+      .map(stepLabelText(for:))
```

Comment:
If you make these `var`s on the enum, I think you could just do `.map(\.stepLabelText)`:

```
var stepLabelText: String {
  switch self {
  case .categorySelection:
    return "Step 1 of 2"
  ... etc
}
```

Reply:
Good call! 👍